registry/coder/modules/digitalocean-region/README.md

```diff
@@ -0,0 +1,155 @@
+---
+display_name: DigitalOcean Region
+description: Provides a region selection parameter for DigitalOcean resources
+icon: ../../../../.icons/digitalocean.svg
+maintainer_github: coder
+verified: true
+tags: [helper, digitalocean, region]
+---
+
+# DigitalOcean Region
+
+This module provides a standardized region selection parameter for DigitalOcean resources. It includes all volume-supporting regions with user-friendly names and country flag icons.
+
+## Features
+
+- **Volume Support**: Only includes regions that support DigitalOcean volumes
+- **User-Friendly**: Display names with country/city information
+- **Visual Icons**: Country flag emojis for each region
+- **Configurable**: Customizable default region and parameter details
+- **Consistent**: Standardized region selection across templates
+
+## Usage
+
+### Basic Usage
+
+```tf
+module "region" {
+  source  = "registry.coder.com/coder/digitalocean-region/coder"
+  version = "1.0.0"
+}
+
+resource "digitalocean_volume" "home_volume" {
+  region = module.region.value
+  # ... other configuration
+}
+
+resource "digitalocean_droplet" "workspace" {
+  region = module.region.value
+  # ... other configuration
+}
+```
+
+### Customized Configuration
+
+```tf
+module "region" {
+  source  = "registry.coder.com/coder/digitalocean-region/coder"
+  version = "1.0.0"
+
+  default      = "sfo3"
+  mutable      = true
+  display_name = "Datacenter Location"
+  description  = "Select the datacenter region for your development environment"
+}
+
+resource "digitalocean_droplet" "workspace" {
+  region = module.region.value
+  # ... other configuration
+}
+```
+
+## Available Regions
+
+The module includes the following DigitalOcean regions:
+
+| Region Code | Location                       | Notes        |
+| ----------- | ------------------------------ | ------------ |
+| `tor1`      | Canada (Toronto)               | 🇨🇦           |
+| `fra1`      | Germany (Frankfurt)            | 🇩🇪           |
+| `blr1`      | India (Bangalore)              | 🇮🇳           |
+| `ams3`      | Netherlands (Amsterdam)        | 🇳🇱 (Default) |
+| `sgp1`      | Singapore                      | 🇸🇬           |
+| `lon1`      | United Kingdom (London)        | 🇬🇧           |
+| `sfo2`      | United States (California - 2) | 🇺🇸           |
+| `sfo3`      | United States (California - 3) | 🇺🇸           |
+| `nyc1`      | United States (New York - 1)   | 🇺🇸           |
+| `nyc3`      | United States (New York - 3)   | 🇺🇸           |
+
+> **Note**: Some regions (nyc1, sfo1, ams2) are excluded because they do not support volumes, which are commonly used for persistent data storage.
```

Comment:
Can we instead add an additional input to control getting only the regiosn with volumes?